Feature/53 rest api geo zones

onetwo1two opened the pull request.

ITsvetkoFF commented on build.gradle:
```diff
@@ -3,6 +3,8 @@ def spring_version = '4.2.5.RELEASE'
 def postgres_version = '9.4.1208.jre7'
 def hsqldb_version = '2.3.3'
+def jackson_version = '2.7.3'
+def model_mapper_version = '0.7.5'
```
it does not deserialize enums
(?)

ITsvetkoFF replied:
Vadym has checked. It does

ITsvetkoFF commented on persistence/src/main/java/edu/softserve/zoo/persistence/repository/impl/GeographicalZoneRepository.java:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.stereotype.Component;
+
+
+@Component
```
Repository

ITsvetkoFF commented on persistence/src/main/java/edu/softserve/zoo/persistence/specification/impl/GeographicalZoneGetAllSpecification.java:
```diff
@@ -0,0 +1,11 @@
+public class GeographicalZoneGetAllSpecification implements HQLSpecification<GeographicalZone> {
+    @Override
+    public String query() {
+        return "from GeographicalZone";
```
getSimpleName

ITsvetkoFF commented on service/src/main/resources/service-ctx.xml:
```diff
@@ -0,0 +1,9 @@
+       xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context.xsd">
+
+    <context:component-scan base-package="edu.softserve.zoo.service.impl"/>
+    <import resource="persistence-ctx.xml"/>
```
ctrl+shift+arrow_up

ITsvetkoFF commented on web/src/main/java/edu/softserve/zoo/controller/rest/GeographicalZoneRestController.java:
```diff
@@ -0,0 +1,39 @@
+ * @author Vadym Holub
+ */
+@RestController
+@RequestMapping("api/v1/geo_zones")
```
to constants

ITsvetkoFF commented on service/src/main/java/edu/softserve/zoo/service/GeographicalZoneService.java:
```diff
@@ -0,0 +1,9 @@
+
+import java.util.Collection;
+
+public interface GeographicalZoneService {
```
common interface

ITsvetkoFF commented on web/src/main/java/edu/softserve/zoo/controller/rest/GeographicalZoneRestController.java:
```diff
@@ -0,0 +1,39 @@
+package edu.softserve.zoo.controller.rest;
```
Spring Test

onetwo1two committed:
added:
- 2 modules (service, web)
- getAll HQL specification for geo_zone
- base service with impl and geo_zone service
- dto
- spring configs
- tests for base and geo_zone service

skyfenko commented on service/src/test/java/edu/softserve/zoo/service/GeographicalZoneServiceImplTest.java:
```diff
@@ -0,0 +1,61 @@
+public class GeographicalZoneServiceImplTest extends CrudServiceImplTest<GeographicalZone> {
+
+    @InjectMocks
+    private GeographicalZoneService geographicalZoneService = new GeographicalZoneServiceImpl();
```
you should not use mocks for services. wire them instead. you have test context

skyfenko commented on service/src/test/java/edu/softserve/zoo/service/GeographicalZoneServiceImplTest.java:
```diff
@@ -0,0 +1,61 @@
+    private GeographicalZoneService geographicalZoneService = new GeographicalZoneServiceImpl();
+
+    @Mock
+    private GeographicalZoneRepository geographicalZoneRepository;
```
wire spring bean
